src/__tests__/LineSegment.test.tsx

```diff
@@ -0,0 +1,92 @@
+import { Point } from "canvas/Geometry/Point";
+import { LineSegment } from "canvas/Geometry/LineSegment";
+import { Vector } from "canvas/Geometry/Vector";
+
+test("a line segment should have a different start and end", async () => {
+    const p1 = new Point(3, 4);
+    const p2 = new Point(3, 4);
+    expect(() => new LineSegment(p1, p2)).toThrow();
+});
+
+test("get length of a line", async () => {
+    const p1 = new Point(0, 0);
+    const p2 = new Point(3, 4);
+    const l = new LineSegment(p1, p2);
+    expect(l.getLength()).toEqual(5);
+});
+
+test("get tangent", async () => {
+    const p1 = new Point(0, 0);
+    const p2 = new Point(3, 4);
+    const l = new LineSegment(p1, p2);
+
+    expect(() => l.getTangent(-0.2)).toThrow();
+    expect(() => l.getTangent(1.2)).toThrow();
+    expect(Vector.changeInAngle(Vector.vectorBetweenPoints(p1, p2), l.getTangent(0))).toEqual(0);
+    expect(Vector.changeInAngle(Vector.vectorBetweenPoints(p1, p2), l.getTangent(0.75))).toEqual(0);
+});
+
+test("split", async () => {
+    const p1 = new Point(0, 0);
+    const p2 = new Point(6, 8);
+    const l = new LineSegment(p1, p2);
+    const lines = l.split(new Point(3, 4));
+
+    expect(lines[0].getStart()).toEqual(l.getStart());
+    expect(lines[1].getEnd()).toEqual(l.getEnd());
+    expect(lines[0].getEnd()).toEqual(lines[1].getStart());
+    expect(lines[0].getLength() + lines[1].getLength()).toEqual(l.getLength());
+});
+
+test("is point near segment", async () => {
+    const l = new LineSegment(new Point(0, 0), new Point(6, 8));
+    const p1 = new Point(0, 0);
+    const p2 = new Point(6, 8);
+    const p3 = new Point(3, 4);
+    const p4 = new Point(3.1, 4);
+    const p5 = new Point(-0.1, 0);
+    const p6 = new Point(6.1, 8);
+    const p7 = new Point(3, 5);
+
+    expect(l.isPointNearSegment(p1, 0.1)).toEqual(p1);
+    expect(l.isPointNearSegment(p2, 0.1)).toEqual(p2);
+    expect(l.isPointNearSegment(p3, 0.1)?.equals(p3)).toBeTruthy();
+    expect(l.isPointNearSegment(p4, 0.1)).toBeTruthy();
+    expect(l.isPointNearSegment(p5, 0.1)).toBeFalsy();
+    expect(l.isPointNearSegment(p6, 0.1)).toBeFalsy();
+    expect(l.isPointNearSegment(p7, 0.1)).toBeFalsy();
+});
+
+test("find intersection point of two lines", async () => {
+    // parallel segments
+    let p1 = new Point(0, 0);
+    let p2 = new Point(3, 4);
+    let p3 = new Point(2, 0);
+    let p4 = new Point(5, 4);
+    let l1 = new LineSegment(p1, p2);
+    let l2 = new LineSegment(p3, p4);
+    expect(LineSegment.findIntersectionPointOfTwoLines(l1, l2, false)).toBeNull();
+    expect(LineSegment.findIntersectionPointOfTwoLines(l1, l2, true)).toBeNull();
+
+    // segments that meet
```

Comment:
Should each of these cases be separate tests?

Reply:
I was trying to keep the number of tests manageable by grouping tests per method. But yeah, that against what we were taught. I could separate everything I guess.

Reply:
I see what you mean. I agree that we should not aim for full coverage, since we have limited time. I am okay with keeping one test per function for now.